apps/web/src/app/layout.tsx

```diff
@@ -8,22 +8,28 @@ import { NextIntlClientProvider } from 'next-intl';
 import { Analytics } from '@vercel/analytics/react';
 import { SpeedInsights } from '@vercel/speed-insights/next';
 import { getLocale, getMessages } from 'next-intl/server';
+import { headers } from 'next/headers';
 import NotificationsHandler from '@/components/misc/notifications-handler';
+import UserLocaleSetter from '@/app/user-locale-setter';
 
 export const metadata: Metadata = {
   title: 'AdsViewer',
   description: 'Get granular insights into your ads performance',
 };
 
 export default async function RootLayout({ children }: { children: React.ReactNode }): Promise<React.ReactNode> {
+  const headersList = headers();
+  const acceptLanguageHeaderValue = headersList.get('accept-language');
+  const preferredLocale = acceptLanguageHeaderValue ? acceptLanguageHeaderValue.split(',')[0] : 'en';
   const locale = await getLocale();
   const messages = await getMessages();
 
   return (
     <html lang={locale}>
```

Comment:
```suggestion
    <html lang={preferredLocale}>
```
Is this correct? If yes we can remove the getLocale function

Reply:
Done